### PR TITLE
UserTagger: fix a number of layout/styling issues

### LINKED ISSUE #2339: User Tag dialog overflows
Whenever it's long enough that the preview wraps.
image

### PR DESCRIPTION
**Pull request by githue**
Fixes #2339

[screenshot below]

Notice the color change for the toggleButton, I think it's better this way because users are used to red meaning off.

### CODE REVIEW

**Comment by githue** [comment]
Update:
usertagger-layout-02

**Comment by githue** [comment]
Travis doesn't like scripts inside hrefs I guess, suggestions?

**Comment by erikdesjardins** [comment]
You could remove the `href` attribute entirely at that point, which has the added benefit of making the link unclickable.

Maybe also:

```css
a:not([href]) {
    color: inherit;
    text-decoration: none;
}
```

**Comment by githue** [comment]
Thank you.

**Comment by githue** [comment]
I have an idea for replacing the color select menu with a chart (using radios hidden inside labels). So users can have more colors to choose from.

https://imgrush.com/O5NzgfQy3GJL

Thoughts?

**Comment by mc10** [comment]
Ehh that's such a cludgy way of matching the selector. Is there a way to avoid the `:not([href])` portion?

**Comment by githue** [comment]
I've decided to move the color selection option to a separate branch because it's not really related to this PR.

Meanwhile, I've added a fix for the user tagger dialog opening outside the page when opened from the side bar. See the moderator list for an example.

**Comment by githue** [comment]
I think this will help with #1125 because it makes it more obvious that the option can be customized.

**Comment by githue** [comment]
Whoops! I forgot to sync master properly before doing the merge and push, so it added unintended commits. How should I resolve this?

**Comment by Ajedi32** [comment]
@githue You could always rebase: `git rebase master` (Then `git log --oneline --graph` to make sure everything looks as expected, and `git push -f` to update the PR.)

File: lib/modules/userTagger.js [hunk 1]
```diff
@@ -137,24 +137,27 @@ addModule('userTagger', function(module, moduleID) {
 			}
 			// add tooltip to document body...
 			userTaggerToolTip = RESUtils.createElement('div', 'userTaggerToolTip', 'RESDialogSmall');
-			var thisHTML = '<h3>Tag User</h3><div id="userTaggerToolTipContents" class="RESDialogContents clear">';
+			var thisHTML = '<h3>Tag User</h3><div id="userTaggerToolTipContents" class="RESDialogContents">';
```
Comment (andytuba) [inline lib/modules/userTagger.js]: I'd love to see this moved into templates.html while you're in here, too.

Comment (githue) [inline lib/modules/userTagger.js]: Done.

File: lib/modules/userTagger.js [hunk 2]
```diff
@@ -225,6 +165,54 @@ addModule('userTagger', function(module, moduleID) {
...
+
+		// Define $tagger.
+		$tagger = $('#userTaggerToolTip');
+		// Add generated elements.
+		$tagger.find('.res-usertag-ignore').append(ignoreToggleButton).append(ignoreSettingsLink);
+		// Populate color selection with options.
+		$colorSelect = $tagger.find('#userTaggerColor');
+		for (var color in bgToTextColorMap) {
+			var bgColor = (color === 'none') ? 'transparent' : color;
+			$colorSelect.append('<option style="background-color: ' + bgColor + '; color: ' + bgToTextColorMap[color] + ' !important;" value="' + color + '">' + color + '</option>');
```
Comment (erikdesjardins) [inline lib/modules/userTagger.js]: You should be able to use [a list](https://mustache.github.io/mustache.5.html#Sections) in the template to accomplish this, i.e. something like (haven't actually tested it):
```html
<select id="userTaggerColor">
  {{#colors}}
    <option style="background-color: {{bgColor}}; color: {{textColor}} !important;" value="{{color}}">{{color}}</option>
  {{/colors}}
</select>
```
and then when you call `template.html()`, pass in an object of this shape:
```js
template.html({
  colors: [
    { bgColor, textColor, color }
  ]
})
```

Comment (andytuba) [inline lib/modules/userTagger.js]: No ES6 in this branch :sob:

```javascript
template.html({
  bgColor: bgColor,
  textColor: textColor,
  color: color
});
```

**Pull request closed without merging**